gce/sysprep/instance_setup.ps1

```diff
@@ -167,7 +160,7 @@ function Change-InstanceName {
   }
 
   $count = 1
-  do {
+  while ($hostname_parts.Length -le 1) {
```

Comment:
I think it's more confusing to use the property of a null object (which would throw an exception in every other language). The do while seems clearer.